pkg/server/v3/server_test.go

```diff
@@ -659,3 +659,108 @@ func TestCallbackError(t *testing.T) {
 		})
 	}
 }
+
+// BENCHMARKS =====================================================================================================
+
+func BenchmarkResponseHandlers(b *testing.B) {
+	for _, typ := range testTypes {
+		b.Run(typ, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				config := makeMockConfigWatcher()
+				config.responses = makeResponses()
+				s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+
+				// make a request
+				resp := makeMockStream(&testing.T{})
+				resp.recv <- &discovery.DiscoveryRequest{Node: node, TypeUrl: typ}
+				go func() {
```

Comment:
Should these goroutines that respond start earlier on? Are we incurring a goroutine startup cost here?

Reply:
I thought about that but I believe if we start the go routine earlier on we'll end up sending on a closed channel which will panic